gluten-core/src/main/scala/io/glutenproject/extension/AddExtraOptimizations.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.extension
+
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+
+case class AddExtraOptimizations(sparkSession: SparkSession) extends (LogicalPlan => Unit) {
+
+  override def apply(plan: LogicalPlan): Unit = {
+    if (!sparkSession.experimental.extraOptimizations.exists(_.isInstanceOf[PullOutProject])) {
+      sparkSession.experimental.extraOptimizations = sparkSession.experimental.extraOptimizations ++
```

Comment:
If I get it corrent, the reason why we add the pull out project rule to `extraOptimizations` is `DecimalAggregates` ? Is there other rules affect ?

Can we inject the pull out project rule to `preCBORules` and apply `DecimalAggregates` first manually ?

Reply:
Yes, currently we have only found that `DecimalAggregates` can cause an impact. I haven't figured out why Spark decides to place `DecimalAggregates` in this position, but it could be because adding `UnscaledValue` might affect the matching of certain optimizers. As a cautious approach, can we handle the expressions added by `DecimalAggregates` in `ColumnarPullOutPreProject`?

Reply:
yes, I think it's better to apply `DecimalAggregates` in `PullOutPreProject` manually at logical side

Reply:
This will cause some issues.
```
Max iterations (100) reached for batch Operator Optimization before Inferring Filters.
```

Reply:
@liujiayi771 are you injecting this rule using `preCBORules` ? it should not affect the Inferring Filters.

Reply:
@ulysses-you Sorry, I used the `OptimizerRules`. Let me try using the `preCBORules` instead.